### PR TITLE
Sick home page

### PR DESCRIPTION
**Pull request by blhohmann**
# Pull Request: Home page revamp

## Description

This pull request adds a prettier home page which is intended to entice users to click. Some of the category images are not the most serious. If that is a problem I can find new ones, but I think it gives our page some charm.

## Changes Made

Home page changes:
- Basically there are 4 cards
- Card 1 is a carousel with each category and a picture, if you click the picture it will send you to the products page filtered by that category (always shows this card)
- Card 2 I call login encouragement. If you aren't logged in, it will say log in to use features and link to that. If you are logged in but don't have a profile it will prompt you to make a profile. If you are logged in and have a profile (every other case) you are told you can update your profile
- Card 3 I call seller registration encouragement. If you are logged in but not a seller, it will say register as a seller to sell stuff. If you are logged in and a seller it will prompt you to make a storefront. If you have a storefront it will prompt you to edit it with template or manual code. In any other case (you aren't logged in) it will prompt you to visit the storefronts index page to see cool storefronts.
- Card 4 always shows, and it just links to your shopping history
 
### To manually test visit home page, and go down the different conditions
4 cards should always show no matter the circumstance, and the correct action should show.

#### To run the rspec tests put this in your terminal
``` bundle exec rspec```
```bundle exec cucumber```

## Impact on Existing Code
- No breaking changes observed

### Include but are not limited to:

Logged out:
image

Logged in && with a profile && not a seller:
image

Logged in && no profile && not a seller:
image

Logged in && profile && seller but no storefront:
image

Logged in && profile && seller with storefront:
image

## Additional Notes

- This feature will need to be extended to work with deals and not just with manually updating price
- No major performance issues have been observed during testing.

## Reviewer Requests

I kindly request a review of these changes. Please provide feedback and suggestions. 1 reviews probably enough.

### Overall Tests
```Coverage report generated for RSpec to /mnt/c/Users/blake/selt2023-final-project-team-001/coverage. 2030 / 2130 LOC (95.31%) covered.```

```
49 scenarios (49 passed)
225 steps (225 passed)
```

## Discussion

Feel free to comment, ask questions, or provide feedback.

Should be no merge conflicts with main.

### CODE REVIEW

File: app/assets/images/home/Collectibles & Art.jpg
Comment (powersaudrey25) [inline app/assets/images/home/Collectibles & Art.jpg:1]: i need this irl

Comment (blhohmann) [reply app/assets/images/home/Collectibles & Art.jpg:1]: fr 😆

**Comment by powersaudrey25** [comment]
A couple things...Whenever I click an image I do not get redirected to that category. Also Update Profile, Use Template, and Code it Yourself don't do anything when clicked. I think we've seen this before and its a turbo issue but I could be wrong. Other than that everything renders correctly and directs as expected

**Comment by blhohmann** [comment]
> A couple things...Whenever I click an image I do not get redirected to that category. Also Update Profile, Use Template, and Code it Yourself don't do anything when clicked. I think we've seen this before and its a turbo issue but I could be wrong. Other than that everything renders correctly and directs as expected

To be clear, that is even when you're running it locally right. On my end when I click the image it redirects :(

**Comment by powersaudrey25** [comment]
> > A couple things...Whenever I click an image I do not get redirected to that category. Also Update Profile, Use Template, and Code it Yourself don't do anything when clicked. I think we've seen this before and its a turbo issue but I could be wrong. Other than that everything renders correctly and directs as expected
> 
> To be clear, that is even when you're running it locally right. On my end when I click the image it redirects :(

Yeah when I run it locally, we can have evan and john try it out too. I remember this exact same thing happening when clicking choose this template a while back too

File: app/views/home/index.html.erb [hunk 1]
```diff
@@ -1 +1,182 @@
...
+          </div>
+        </div>
+      </div>
+    <% elsif @user.profile.present?%>
+      <div class="col-lg-3">
+        <div class="card">
+          <div class="card-body d-flex flex-column justify-content-between">
+            <h5 class="card-title">Update your profile</h5>
+            <p class="card-text">Let others know more about yourself</p>
+            <a href="<% "profiles/#{@user.profile.id}/edit" %>" class="btn btn-primary mt-auto">Update profile</a>
```
Comment (evfinkn) [inline app/views/home/index.html.erb:97]:
```suggestion
            <a href="<%= "profiles/#{@user.profile.id}/edit" %>" class="btn btn-primary mt-auto">Update profile</a>
```

File: app/views/home/index.html.erb [hunk 2]
```diff
@@ -1 +1,182 @@
...
+          </div>
+        </div>
+      <% elsif @user && @user.storefront %>
+        <div class="col-lg-3">
+          <div class="card">
+            <div class="card-body d-flex flex-column justify-content-between">
+              <h5 class="card-title">Make your Storefront cooler</h5>
+              <p class="card-text">Use a premade template, or code it yourself!</p>
+              <a href="<%"/storefronts/#{@user.storefront.id}/choose_template"%>" class="btn btn-primary mt-auto">Use template</a>
+              <a href="<%"/storefronts/#{@user.storefront.id}/customize"%>" class="btn btn-primary mt-auto">Code it yourself</a>
```
Comment (evfinkn) [inline app/views/home/index.html.erb:140]:
```suggestion
              <a href="<%="/storefronts/#{@user.storefront.id}/choose_template"%>" class="btn btn-primary mt-auto">Use template</a>
              <a href="<%="/storefronts/#{@user.storefront.id}/customize"%>" class="btn btn-primary mt-auto">Code it yourself</a>
```

**Review by evfinkn — COMMENTED** [review]
This should fix the buttons not working (fingers crossed).

**Comment by blhohmann** [comment]
> This should fix the buttons not working (fingers crossed).

Any thoughts on why clicking image works for me but not Audrey? Idk if that's turbo related or something else.

File: app/views/home/index.html.erb [hunk 3]
```diff
@@ -1 +1,182 @@
...
+          <p class="card-text">See what you've viewed</p>
+          <a href="/history" class="btn btn-primary mt-auto">View history</a>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
+
+<script>
+    document.addEventListener("DOMContentLoaded", function() {
```
Comment (evfinkn) [inline app/views/home/index.html.erb:168]:
```suggestion
    document.addEventListener("turbo:load", function() {
```

**Review by evfinkn — COMMENTED** [review]
This fixes the pictures not linking (which is happening for me as well).

File: app/views/home/index.html.erb (hunk 3 above)
Comment (blhohmann) [reply app/views/home/index.html.erb:168]: Thanks you're the goat. I wonder if locally mine isn't using turbo or something...

File: app/views/home/index.html.erb [hunk 4]
```diff
@@ -1 +1,182 @@
...
+          <!--from https://www.youtube.com/watch?app=desktop&v=bmFXJI-F8hw-->
+          <!--from https://fatphobiabusters.tumblr.com/post/157054882853/the-boss-baby-wtf-are-we-fat-shaming-babies-->
+          <!--from https://knowyourmeme.com/memes/cat-crunching-crunchy-cat-->
+
+
+          <% categories.each_with_index do |category, index| %>
+            <!--want this in case later categories are added, but the pictures aren't-->
+            <% if Rails.application.assets.find_asset("home/#{category}") %>
+              <div class="carousel-item <%= 'active' if index.zero? %>">
+                <img src="<%= asset_path("home/#{category}.jpg") %>" class="d-block carousel-image" alt="<%= category %>">
```
Comment (evfinkn) [inline app/views/home/index.html.erb:62]: You can actually replace the script altogether if you just add the link to the image directly:
```suggestion
                <%= link_to products_path(category: index + 1, order: "desc") do %>
                  <img src="<%= asset_path("home/#{category}.jpg") %>" class="d-block carousel-image" alt="<%= category %>">
                <% end %>
```

Comment (blhohmann) [reply app/views/home/index.html.erb:62]: That's facts. I'll do that after my hw due tonight.

**Comment by powersaudrey25** [comment]
Working as expected! Thanks! Great work love how interactive it is

**Review by powersaudrey25 — APPROVED** [review]

**Pull request merged**